src/Card/CardFooter/CardFooter.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useResizeDetector } from 'react-resize-detector/build/withPolyfill';
+
+import { useReversedChildren } from '../../hooks';
+
+export const CardFooterContext = React.createContext();
+
+export const CARD_FOOTER_BREAKPOINTS = {
+  STACKED: 460,
+};
+
+const CardFooter = ({
+  className,
+  children,
+  stacked: controlledStacked,
+  breakpoints,
+  ...attrs
+}) => {
+  const [stacked, setStacked] = useState(controlledStacked);
+
+  const handleResize = (width) => {
+    if (controlledStacked) {
+      // ``stacked`` was passed in as a prop so this component is being used as a
+      // controlled component; rely on this prop instead of setting ``stacked`` state below.
+      return;
+    }
+
+    // TODO: what to do about hardcoded width breakpoints...?
+    if (width < (breakpoints?.stacked || CARD_FOOTER_BREAKPOINTS.STACKED)) {
+      setStacked(true);
+    } else {
+      setStacked(false);
+    }
+  };
+
+  const { ref } = useResizeDetector({
+    handleHeight: false,
+    refreshMode: 'throttle',
+    refreshRate: 100,
+    onResize: handleResize,
+  });
```

Comment:
I wonder the performance implication of this with 100+ cards on screen

Reply:
Me too. Supposedly `ResizeObservers` are natively quite performant, especially when throttled. This is the most sane approach I could cobble together quickly to let the component be responsive based on its own element width rather than a media query (screen width) as `Cards` take up the full width of their container.

CSS will eventually have container queries for this use case but they are not widely supported as of now, so this sort of approach using `ResizeObservers` seems to be the next best strategy.

I'd also hope we're not showing 100+ cards on a screen at once 😉 If we had a use case for that, we might only render the `Card` components into the DOM that are in the user's viewport. Regardless, it'd worth testing the performance of this out more scientifically.